### PR TITLE
Ensure computations’ stdout JSON errors are human-readable

### LINKED ISSUE #49: Improve simulator’s JSON-parsing error
_coinstac-simulator_ errors when a command-computation’s stdout content isn’t JSON. It doesn’t output meaningful information about the error, just the standard `JSON.parse` error. Algorithm authors don't necessarily have the skills to run their computations via CLI, and they definitely don't know how to dig throw COINSTAC internals and debug. We need to add some meaningful error output, like:

```
  29: [
  30:   "whatcha": doin,
                   ^
                   Invalid here!
```

### PR DESCRIPTION
**Pull request by swashcap**
### Problem

Errors in parsing computations’ stdout as JSON aren’t clearly visible to authors (see #49)
### Solution

Use [json-parse-helpfulerror](https://www.npmjs.com/package/json-parse-helpfulerror). Clean up the method and tests. Bump coverage to actual 100%.
### Testing
1. Check out the _enhancement/#49-command-json-error_ branch in your local repo
2. Run `npm run bootstrap` from the repo’s root to install new dependencies
3. Run `npm link` in _packages/coinstac-simulator_ to globally link the package
4. Create the following files in a test directory:
   
   ``` js
   // computation.js
   module.exports = {
     name: 'my-bad-thingy',
     version: '1.0.0',
     local: {
       type: 'cmd',
       cmd: 'node',
       args: ['./hello-dude.js'],
       verbose: true,
     },
     remote: {
       type: 'fn',
       fn: function(params) {
         console.log(params);
         return { complete: true };
       },
     },
   };
   ```
   
   ``` js
   // declaration.js
   module.exports = {
     computationPath: './computation.js',
     local: Array(1),
     verbose: true,
   };
   ```
   
   ``` js
   // hello-dude.js
   console.log(`[{
     "hello": "dude",
     "what": "is up"
   }, {
     "not": "much"
   }, {
     "yolo": 88,
   }]`);
   ```
5. Run `coinstac-simulator -d declaration.js` in your test directory
6. Observe easy-to-understand JSON parsing error

Closes #49.

### CODE REVIEW

**Review by rssk — CHANGES_REQUESTED** [review]

File: packages/coinstac-common/src/models/computation/command-computation.js [hunk 1]
```diff
@@ -28,61 +30,71 @@ class CommandComputation extends Computation {
...
-          /* eslint-enable */
 
-          /* istanbul ignore if */
-          if (this.verbose) { console.error(errMsg); } // eslint-disable-line
-          return rej(err);
-        }
-        return res(rslt);
-      });
-      return null;
+          // ?
```
[the remark this answers is not in the document]
Comment (swashcap) [reply packages/coinstac-common/src/models/computation/command-computation.js]: The variables retain old state unless you reset them. No idea why.

File: packages/coinstac-common/test/models/computation/command.js [hunk 2]
```diff
@@ -16,113 +18,204 @@ test('model::CommandComputation constructor', t => {
...
+  const command = getStubbedCommand();
+  const cmd = 'oh-my-executables';
+  const cwd = 'rando-directory';
+  const opts = {
+    will: 'this make',
+    some: 'pretty json?',
+  };
+  const spawnStub = sinon.stub(cp, 'spawn').returns(command);
+  const stderr = [
+    'a penny for ',
```
[the remark this answers is not in the document]
Comment (swashcap) [reply packages/coinstac-common/test/models/computation/command.js:39]: 💰

File: packages/coinstac-common/test/models/computation/command.js [hunk 3]
```diff
@@ -16,113 +18,204 @@ test('model::CommandComputation constructor', t => {
...
-  .catch(() => {
-    t.ok('handled exception');
-    return t.end();
+tape('CommandComputation#run verbose logging', t => {
+  const args = ['some', 'silly', 'arguments'];
+  const cmd = 'macho';
+  const command = getStubbedCommand();
+
+  /**
+   * @todo Don't spy on console methods! Figure out how to inject the logger and
```
[the remark this answers is not in the document]
Comment (swashcap) [reply packages/coinstac-common/test/models/computation/command.js:103]: I believe tape itself uses `console.log` and `console.error`. Makes things confusing. We should inject the logger instance, anyhow.

File: packages/coinstac-common/src/models/computation/command-computation.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rssk) [reply packages/coinstac-common/src/models/computation/command-computation.js]: Mind putting in a more descriptive comment?

File: packages/coinstac-common/test/models/computation/command.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rssk) [reply packages/coinstac-common/test/models/computation/command.js:103]: Make sense 👍

File: packages/coinstac-common/src/models/computation/command-computation.js (hunk 1 above)
Comment (swashcap) [inline packages/coinstac-common/src/models/computation/command-computation.js]: Added in 8d7181e3c863fd74e5a07b4861c609092bccb68f.

[the remark this answers is not in the document]
Comment (swashcap) [reply packages/coinstac-common/src/models/computation/command-computation.js]: Added in 8d7181e3c863fd74e5a07b4861c609092bccb68f.

**Review by rssk — APPROVED** [review]

**Pull request merged**